include/jemalloc/internal/cache_bin.h

```diff
@@ -294,6 +298,21 @@ cache_bin_alloc(cache_bin_t *bin, bool *success) {
 	return cache_bin_alloc_impl(bin, success, true);
 }
 
+JEMALLOC_ALWAYS_INLINE cache_bin_sz_t
+cache_bin_alloc_batch(cache_bin_t *bin, size_t num, void **out) {
+	size_t n = cache_bin_ncached_get_internal(bin);
+	if (n > num) {
+		n = num;
+	}
+	memcpy(out, bin->stack_head, n * sizeof(void *));
+	bin->stack_head += n;
+	uint16_t head_low_bits = (uint16_t)(uintptr_t)bin->stack_head;
+	if (head_low_bits > bin->low_bits_low_water) {
```

Comment:
This check is not necessarily correct in the case where a bin spans a 64k boundary. I think, given that we don't need to save every cycle on the batch paths, it's fine to compute the low_water_mark position explicitly.

Reply:
It feels we gotta to better make sure the low_bits fields have only equality check against them. Maybe similar to the TSD, i.e. change the member names to something tedious to type, and then provide wrappers that we must always go through.